lib/cli-engine/cli-engine.js

```diff
@@ -785,7 +786,9 @@ class CLIEngine {
         // Iterate source code files.
         for (const { config, filePath, ignored } of fileEnumerator.iterateFiles(patterns)) {
             if (ignored) {
-                results.push(createIgnoreResult(filePath, cwd));
+                if (warnIgnored) {
+                    results.push(createIgnoreResult(filePath, cwd));
+                }
                 continue;
             }
```

Comment:
This might seem odd, but
if `ignored = false, warnIgnored = false` -> lint
if `ignored = false, warnIgnored = true` -> lint
if `ignored = true, warnIgnored = false` -> do not show warning, and do not lint
if `ignored = true, warnIgnored = true` -> show warning, do not lint